cake-autorate.sh

```diff
@@ -32,26 +30,23 @@ cleanup_and_killall()
 	log_msg "INFO" ""
 	log_msg "INFO" "Killing all background processes and cleaning up temporary files."
 
-	if ! [[ -z ${maintain_pingers_pid} ]]; then
+	if [[ -n ${maintain_pingers_pid} ]]; then
 		log_msg "DEBUG" "Terminating maintain_pingers_pid: ${maintain_pingers_pid}."
 		kill_and_wait_by_pid_name maintain_pingers_pid 0
 	fi
 
-	if ! [[ -z ${monitor_achieved_rates_pid} ]]; then
+	if [[ -n ${monitor_achieved_rates_pid} ]]; then
 		log_msg "DEBUG" "Terminating monitor_achieved_rates_pid: ${monitor_achieved_rates_pid}."
 		kill_and_wait_by_pid_name monitor_achieved_rates_pid 0
 	fi
 
-	if ! [[ -z ${maintain_log_file_pid} ]]; then
+	if [[ -n ${maintain_log_file_pid} ]]; then
 		log_msg "DEBUG" "Terminating maintain_log_file_pid: ${maintain_log_file_pid}."
 		kill_and_wait_by_pid_name maintain_log_file_pid 0
 	fi
 
 	[[ -d ${run_path} ]] && rm -r ${run_path}
-	[[ -d /var/run/cake-autorate ]] && compgen -G /var/run/cake-autorate/* > /dev/null || rm -r /var/run/cake-autorate
-
-	# close sleep file descriptor
-	exec {sleep_fd}>&-
+	rmdir /var/run/cake-autorate 2>/dev/null
```

Comment:
I like the idea of using 'rmdir' in this way.

Reply:
So no changes needed?

Reply:
Yes much more elegant than my crazy compgen idea.